proposals/2117-marquez-over-time.md

```diff
@@ -0,0 +1,34 @@
+# Proposal: Marquez API to retrieve historical data
+
+## Problem
+
+Marquez data model in PostgresSQL allows extracting a snapshot of any lineage content from the past. This may be extremely
+useful to track how job, dataset etc. evolved over time. The feature can be implemented within each Marquez endpoint: like
+retrieving datasets or jobs. The purpose of this doc is to agree on a consistent API implementation approach. 
+
+## Solution
+
+Point in time can be described by a datetime, a dataset version or a job version. In case of dataset version or job version, an API should find out 
+a point time correlated with dataset or job version (like the `created_at` of `dataset_versions` row) and prepare a snapshot based 
+on the entries older that. 
+
+Examples:
+
+ * `/api/v1/namespaces/some-namespace/datasets/some-dataset?snapshotAt=datetime:2011-12-03T10%3A15%3A30%2B01%3A00` which is an url encoded version of `ZonedDateTime` `2011-12-03T10:15:30+01:00`
+ * `/api/v1/namespaces/some-namespace/datasets/some-dataset?snapshotAt=dataset_version:5ca3b37e-4e18-11ed-bdc3-0242ac120002`
+ * `/api/v1/namespaces/some-namespace/jobs/some-job?snapshotAt=job_version:5ca3b37e-4e18-11ed-bdc3-0242ac120002`
+ 
+Please mind that requested version id does not have to be an id of the requested dataset. A user debugging issue 
+with `dataset_x` at version `ver_x`, can request `dataset_y` snapshot at `ver_x`. The request should be still valid. 
```

Comment:
This only works if the dataset version is a strict UUID. In https://github.com/MarquezProject/marquez/pull/2153 , we're proposing using external version identifiers, which may not be globally unique.

Reply:
ok, let's modify the assumption and remove such feature.